libs/core/src/lib/file-uploader/file-uploader.component.ts

```diff
@@ -0,0 +1,268 @@
+
+    /** @hidden */
+    onChange: Function = () => { };
+
+    /** @hidden */
+    onTouched: Function = () => { };
+
+
+
+    /** @hidden */
+    registerOnChange(fn: any): void {
+        this.onChange = fn;
+    }
+
+    /** @hidden */
+    registerOnTouched(fn: any): void {
+        this.onTouched = fn;
+    }
+
+    /** @hidden */
+    setDisabledState(isDisabled: boolean): void {
+        this.disabled = isDisabled;
+        this.changeDetRef.detectChanges();
+    }
+
+    /** @hidden */
+    writeValue(files: File[]): void {
+        // not needed - should be handled by user.
+    }
+
+    /** @hidden */
+    selectHandler(event: File[]): void {
+
+        this.validateFiles(event);
+        if (this.validFiles.length > 0) {
+            this.setInputValue(this.validFiles);
+            this.onChange(this.validFiles);
+            this.selectedFilesChanged.emit(this.validFiles);
+        }
+        if (this.invalidFiles.length > 0) {
+            this.selectedInvalidFiles.emit(this.invalidFiles);
+        } else {
+            this.setInputValue(event);
+            this.onChange(event);
+            this.selectedFilesChanged.emit(event);
+        }
+
+    }
+
+    validateFiles(event: File[]): void {
+        if (this.fileLimit && event.length > this.fileLimit) {
+            throw new Error('FileLimitError - Selected files count is more than specified limit ');
+        }
+        if (this.maxFileSize && this.minFileSize) {
+            const maxsize = this.parseFileSize(this.maxFileSize);
+            const minsize = this.parseFileSize(this.minFileSize);
+            event.forEach((file) => {
+                if (file.size < maxsize && file.size > minsize) {
+                    this.validFiles.push(file);
+                } else {
+                    this.invalidFiles.push(file);
+                }
+            });
+        } else if (this.maxFileSize) {
+            const maxsize = this.parseFileSize(this.maxFileSize);
+            event.forEach((file) => {
+                if (file.size < maxsize) {
+                    this.validFiles.push(file);
+                } else {
+                    this.invalidFiles.push(file);
+                }
+            });
+        } else if (this.minFileSize) {
+            const minsize = this.parseFileSize(this.minFileSize);
+            event.forEach((file) => {
+                if (file.size > minsize) {
+                    this.validFiles.push(file);
+                } else {
+                    this.invalidFiles.push(file);
+                }
+            });
+        }
+
+    }
+
+    setInputValue(selectedFiles: File[]): void {
+        let fileName = '';
+        selectedFiles.forEach((file) => {
+            fileName = fileName.concat(' ' + file.name);
+        });
+        this.inputRefText.nativeElement.value = fileName;
+        this.inputRefText.nativeElement.title = fileName;
+    }
+
+    parseFileSize(fileSize: string): Number {
```

Comment:
Let's check if fileSize is empty and return null or 0.
Also let's change complex `Number` to simple `number`

Reply:
Changes are done.